i18n-helpers/src/lib.rs

```diff
@@ -477,6 +477,40 @@ fn heuristic_codeblock<'a>(
     Ok((groups, ctx))
 }
 
+/// Special Admonish "codeblock" - extract the body content for translation
+/// If a title is present, also extract that for translation
+/// Note this is from mdbook-admonish: https://github.com/tommilligan/mdbook-admonish
+fn admonish_codeblock<'a>(
+    events: &'a [(usize, Event<'_>)],
+    mut ctx: GroupingContext,
+) -> Result<(Vec<Group<'a>>, GroupingContext), CmarkError> {
+    // First, ensure we're dealing with a code block
+    if events.len() < 2
+        || !matches!(&events[0].1, Event::Start(Tag::CodeBlock(_)))
+        || !matches!(&events[events.len() - 1].1, Event::End(TagEnd::CodeBlock))
+    {
+        return heuristic_codeblock(events, ctx);
+    }
+
+    // Check if we're dealing with a fenced code block
+    if !matches!(
```

Comment:
The `matches!` macro is not typically your first tool. Constructs like `match`, `if let`, and the new `let patter = expression else { return ... }` are your friend.